### PR TITLE
change TSI socket max frame size to 16KB

### PR DESCRIPTION
**Pull request by yihuazhang**
This PR resolves the first issue described in https://github.com/envoyproxy/envoy/issues/15296. 

Note that we need to set the maximum frame size (16KB) in the following two places: 

1. When creating an ALTS TSI handshaker 
2. When creating a zero-copy frame protector

The frame size set in 1. will be used in the frame size negotiation by telling its peer what is the desired frame size for the sender. The frame size set in 2. will be used to compare with the peer's frame size to determine the final frame size to use. 
We need to ensure the same frame size is used in 1. and 2. 

Notice that both API's used in 1. and 2. are gRPC TSI surface API's (alts_tsi_handshaker_create() and  tsi_handshaker_result_create_zero_copy_grpc_protector()) so all we need to ensure is to pass the same frame size to those API's. After this PR is merged, I will prepare for a patch to update the max frame size used in the internal TSI handshaker.

### CODE REVIEW

**Comment by yihuazhang** [comment]
@antoniovicente F.Y.I.

File: source/extensions/transport_sockets/alts/config.cc
Comment (antoniovicente) [inline source/extensions/transport_sockets/alts/config.cc:129]: Thanks for making these changes.

Would it be possible to add some basic tests to cover the behavior changes in this PR?

Comment (htuch) [reply source/extensions/transport_sockets/alts/config.cc:129]: +1 on tests. Thanks @yihuazhang for the quick PR!

Comment (yihuazhang) [reply source/extensions/transport_sockets/alts/config.cc:129]: It took some time figuring out how the current envoy/alts tests work. I added some tests to verify if a correct frame size (16KB) will be used by a protector. There are some tests that I want to add, which depend on the change in gRPC code base (https://github.com/grpc/grpc/pull/25621). I added TODO for them, and will fix them once the PR is in and envoy updates the gRPC version.

File: source/extensions/transport_sockets/alts/tsi_socket.cc
Comment (antoniovicente) [inline source/extensions/transport_sockets/alts/tsi_socket.cc:136]: The formatter really wants you to do a few changes, please look at:

https://dev.azure.com/cncf/envoy/_build/results?buildId=68287&view=logs&j=55bc2ab9-36c1-5ab8-58a6-8c258728eeff&t=33b57bf9-ef13-58c8-bc2b-f804aeb16d39

**Review by antoniovicente — COMMENTED** [review]
Change looks great, but needs a format fix.

**Comment by htuch** [comment]
@yihuazhang thanks for the fix! I had a quick look at the test and one thing I was wondering about is how much of the bug is fixed here before we do a TSI library update. Is this just changing the Envoy max frame size plumbing, but we need a TSI-side fix as well for it to be fully effective?

File: source/extensions/transport_sockets/alts/tsi_socket.cc
Comment (yihuazhang) [reply source/extensions/transport_sockets/alts/tsi_socket.cc]: Done.

**Comment by yihuazhang** [comment]
@htuch The changes in this PR does not need the TSI library change to become affective. The TSI library change is just for a more comprehensive testing for the changes in this PR. Once this PR is in, we should be able to resolve the first issue mentioned in https://github.com/envoyproxy/envoy/issues/15296. Still two more to go :)

**Review by htuch — APPROVED** [review]
LGTM, will hand this over to @antoniovicente for final approval/merge as CODEOWNER.

**Comment by antoniovicente** [comment]
/retest

multiarch is known to be flaky/broken.

**Comment by antoniovicente** [comment]
Can you merge upstream/main ?

I don't know what's up with the docker linux multiarch failure.  Hopefully merging upstream/main will fix it.

/wait

**Comment by yihuazhang** [comment]
It seems the same test still fails even after merging with upstream/main :(

**Comment by yihuazhang** [comment]
@antoniovicente,  so I guess we need to wait until https://github.com/envoyproxy/envoy/issues/14971 is resolved to merge this PR?

**Comment by antoniovicente** [comment]
> @antoniovicente, so I guess we need to wait until #14971 is resolved to merge this PR?

It may be possible for someone with superuser permissions like @htuch to force the merge.  Let's see if the most recent retry succeeds.

**Pull request merged**
